tests/sparseml/pytorch/recipe_template/test_end_to_end.py

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from subprocess import PIPE, run
+
+from sparseml import recipe_template
+
+
+def test_cli_entrypoint_invocation():
+    output = run(["sparseml.recipe_template"], stdout=PIPE, stderr=PIPE)
+    assert "EpochRangeModifier" in output.stdout.decode()
+    assert "LearningRateFunctionModifier" in output.stdout.decode()
+
+
+def test_function_entrypoint():
+    recipe_template()
```

Comment:
This saves something to recipe.md right? can you change filename to tmp path?

Reply:
Yes, updated the logic to make testing easier

Reply:
The upcoming changes need a filename to be specified to write a recipe, so we're good here